pkg/sql/sem/tree/select.go

```diff
@@ -1199,6 +1199,37 @@ func (p LockingWaitPolicy) Max(p2 LockingWaitPolicy) LockingWaitPolicy {
 	return LockingWaitPolicy(max(byte(p), byte(p2)))
 }
 
+// LockingDurability represents the durability of a lock. It is currently not
+// exposed through SQL, but is instead set according to statement type and
+// isolation level. It is included here for completeness.
+type LockingDurability byte
+
+const (
+	// LockDurabilityBestEffort represents the default: make a best-effort attempt
+	// to hold the lock until commit while keeping it unreplicated and
+	// in-memory. This must not be used when correctness depends on locking.
+	LockDurabilityBestEffort LockingDurability = iota
+
+	// LockDurabilityGuaranteed guarantees that if the transaction commits, the
+	// lock was held until commit. This must be used when correctness depends on
+	// locking.
+	LockDurabilityGuaranteed
+)
+
+var lockingDurabilityName = [...]string{
+	LockDurabilityBestEffort: "best-effort",
+	LockDurabilityGuaranteed: "guaranteed",
```

Comment:
(I'm not attached to these names, so please feel free to suggest different ones if you don't like these.)